syntax = "proto3";

// filePath

// fileName

package Message;

message Reply {
  bool ret = 1;
}

message MessageInfo {
  string timestamp = 1;
  uint32 carId = 2;
}
message MessageEntry {
  string timestamp = 1;
  uint32 carId = 2;
  uint32 x = 3;
  uint32 y = 4;
  uint32 v_x = 5;
  uint32 v_y = 6;
  uint32 v_r = 7;
  uint32 direction = 8;
  string img = 9;
}

// The greeting service definition.
service TSDB {
  // set in the tsdb
  rpc TsSet(MessageEntry) returns (Reply) {};
  // get from the tsdb
  rpc TsGet(MessageInfo) returns (MessageEntry) {};
}
